janus_core/single_point.py

```diff
@@ -179,7 +181,59 @@ def _get_stress(self) -> MaybeList[ndarray]:
 
         return self.struct.get_stress()
 
-    def run_single_point(self, properties: MaybeSequence[str] = ()) -> CalcResults:
+    @staticmethod
+    def _remove_invalid_props(
+        struct: Atoms,
+        properties: Optional[list[str]] = None,
+    ) -> None:
+        """
+        Remove any invalid properties from calculator results.
+
+        Parameters
+        ----------
+        struct : Atoms
+            Structure with attached results from calculator.
+        properties : Optional[list[str]]
+            Physical properties requested to be calculated. Default is [].
+        """
+        properties = properties if properties else []
+        rm_keys = []
+
+        # Find any properties with non-finite values
+        for prop in struct.calc.results:
+            if not isfinite(struct.calc.results[prop]).all():
+                rm_keys.append(prop)
+        # Raise error if property was explicitly requested, otherwise remove
+        for prop in rm_keys:
+            if prop in properties:
+                raise ValueError(
+                    f"'{prop}' contains non-finite values for this structure."
+                )
+            struct.calc.results.pop(prop)
```

Comment:
Do you think this gives better intent as:
```suggestion
            del struct.calc.results[prop]
```
?